webapi/src/main/java/org/sakaiproject/webapi/controllers/CardGameController.java

```diff
@@ -332,19 +332,33 @@ private String[] getAllowedRoles() {
         return configuredAllowedRoles != null ? configuredAllowedRoles : ALLOWED_ROLE_IDS_DEFAULT;
     }
 
-    private Boolean hasDefaultImage(String userId, String siteId, boolean official) {
+    private Boolean hasDefaultImage(String userId, String siteId, boolean preferOfficial) {
         if (StringUtils.isAnyBlank(userId, siteId)) {
             log.debug("blank userId or siteId");
             return null;
         }
 
-        ProfileImage profileImage;
-        if (official) {
-            profileImage = profileImageLogic.getOfficialProfileImage(userId, siteId);
+        ProfileImage primaryImage;
+        ProfileImage fallbackImage = null;
+        
+        if (preferOfficial) {
+            // Try official image first
+            primaryImage = profileImageLogic.getOfficialProfileImage(userId, siteId);
+            
+            // If official image is default/initials, try profile image as fallback
+            if (primaryImage.isDefault() || primaryImage.isInitials()) {
+                fallbackImage = profileImageLogic.getProfileImage(userId, null, ProfileConstants.PROFILE_IMAGE_MAIN, siteId);
+            }
         } else {
-            profileImage = profileImageLogic.getProfileImage(userId, null, ProfileConstants.PROFILE_IMAGE_MAIN, siteId);
+            // Use profile image directly
+            primaryImage = profileImageLogic.getProfileImage(userId, null, ProfileConstants.PROFILE_IMAGE_MAIN, siteId);
         }
-
-        return profileImage.isDefault() || profileImage.isInitials();
+        
+        // Check if we have a usable image (either primary or fallback)
+        boolean primaryIsDefault = primaryImage.isDefault() || primaryImage.isInitials();
+        boolean fallbackIsDefault = fallbackImage == null || fallbackImage.isDefault() || fallbackImage.isInitials();
+        
+        // Only skip user if both primary and fallback (if applicable) are default/initials
+        return primaryIsDefault && fallbackIsDefault;
```

Comment:
seems like this check should be in the service and the controller is simplified, benefit is that any code calling the service method would benefit.